custom-elements/reactions/resources/reactions.js

```diff
@@ -0,0 +1,213 @@
+
+let testNumber = 1;
+
+function defineNewCustomElement(observedAttributes) {
+    let log = [];
+    let name = 'custom-element-' + testNumber++;
+
+    class CustomElement extends HTMLElement {
+        constructor() {
+            super();
+            log.push({type: 'constructed', element: this});
+        }
+        attributeChangedCallback(name, oldValue, newValue) { log.push({type: 'attributeChanged', element: this, name: name, oldValue: oldValue, newValue: newValue}); }
```

Comment:
I guess it's probably worth adding namespace here too, even if it's a very unloved argument.